File: csharp/bob/Bob.cs
using System;
using System.Text.RegularExpressions;

public static class Bob
{
    public static string Response(string statement)
    {
      bool isSilence = !Regex.IsMatch(statement, @"[A-Za-z0-9?]+") || statement == String.Empty;
      bool isYelling = statement.ToUpper() == statement && Regex.IsMatch(statement, @"[A-Z]+");
      bool isQuestion = statement.Trim().EndsWith("?");

      if(isSilence)
        return "Fine. Be that way!";
      if(isYelling)
      {
        if(isQuestion) return "Calm down, I know what I'm doing!";
        return "Whoa, chill out!";
      }
      if(isQuestion)
        return "Sure.";
      return "Whatever.";
    }
}

File: javascript/bob/bob.js
module.exports = class Bob {
  constructor() {}

  hey(str) {
    str = str.trim();
    if(/[A-Z]+/.test(str) && str.toUpperCase() === str) return "Whoa, chill out!";
    if(str.substring(str.length - 1) === "?") return "Sure.";
    if(str === '') return "Fine. Be that way!";
    return "Whatever.";
  }
}
